app/controllers/spaces_controller.rb

```diff
@@ -8,4 +8,33 @@ def show
     @space = Space.find(params[:id])
 
   end
+
+  def new
+    @space = Space.new()
+  end
+
+  def create
+    @user = current_user
+    @space = Space.new(space_params)
+    @space.user = @user
+    @space.save ? (redirect_to space_path(@space)) : (render :new)
+  end
+
+  def edit
+    @space = Space.find(params[:id])
+  end
+
+  def update
+    @space = Space.find(params[:id])
+    @space.update(space_params) ? (redirect_to space_path(@space)) : (render :edit)
+  end
+
+
+
+
+  private
+
+  def space_params
+    params.require(:space).permit(:name,:address,:price, :capacity)
```

Comment:
Add in spaces betweer the parameters in .permit:
.permit(:name, :address, :price, :capacity)